app/src/main/java/team7/voluntime/Fragments/Common/UserProfileFragment.java

```diff
@@ -1,42 +1,148 @@
 package team7.voluntime.Fragments.Common;
 
 
-import android.content.Intent;
-import android.os.Bundle;
 import android.app.Fragment;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
 import android.support.annotation.Nullable;
-import android.util.Log;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+import android.widget.Toast;
 
 import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ValueEventListener;
 
+import butterknife.BindView;
 import butterknife.ButterKnife;
 import butterknife.OnClick;
-import team7.voluntime.Activities.LoginActivity;
 import team7.voluntime.Activities.MainActivity;
 import team7.voluntime.R;
 
+import static android.view.View.VISIBLE;
+
 /**
  * A simple {@link Fragment} subclass.
  */
 public class UserProfileFragment extends Fragment {
 
-    private final static String TAG = "MapFragment";
+    private final static String TAG = "UserProfileFragment";
+
+    //Local variables used in the store database values
+    private String Type;
+    private String UserName;
+    private String DOB;
+    private String Phone;
+    private String Address;
+    @BindView(R.id.charityStuff)
+    LinearLayout charityStuff;
+    @BindView(R.id.uCat)
+    TextView uCat;
+    @BindView(R.id.uDesc)
+    TextView uDesc;
```

Comment:
what is uCat and uDesc? 
Could we use more descriptive naming for our ids? Maybe what i currently do is overkill but just not sure what these are meant to be.  
I like to namespace ids with the class as well, again, might be overkill but at least it means each id can be referenced to the class.